### DISCUSSION TITLE
Windows x64 openssl version 2.2 is not working, error 0x80004002

### DISCUSSION THREAD

**Discussion by MinwooWebeng** [microsoft/msquic#3999] [Q&A] [answered]
Windows x64 openssl binary seems not working on windows 10 pro.

Steps I took: 
1. download msquic 2.2 binary release, unzip
2. visual studio 2022 community edition, new c++ empty project
3. copy 2.2 release folder into project directory and setup dependency(additional includes and linker input, I used static library)
5. create main source file with provided sample
6. build and run. It works fine (prints the default message)
7. go to binary folder with powershell
8. generate server certificate and key with the openssl command provided by the sample file
10. .\quictest.exe -server -cert_file:server.cert -key_file:server.key
11. prints 'ConfigurationLoadCredential failed, 0x80004002!'

also, the client does not work. 

I read https://github.com/microsoft/msquic/issues/1092,
but It seems using schannel. I'm using openssl.

Can I get some help running msquic?

**Comment by nibanks** [comment]
That error indicates `QUIC_STATUS_NOT_SUPPORTED` so you're trying to do something that isn't supported. I recommend you grab the logs to see what's up here: https://github.com/microsoft/msquic/blob/main/docs/Diagnostics.md#trace-collection.

**Reply by MinwooWebeng** [reply]
Thanks for your quick response!

I checked the document. Unfortunately, the netsh.exe seemed not working for me.
It prints  명령 옵션 "globalKeywords"에 대한 값 "-outputFormat"이(가) 잘못되었습니다.
자세한 내용은 '추적 start 도움말'을 참조하십시오. 

(sorry for my desktop language setting. It says command option 'globalKeywords' value '-outputFormat' is wrong.)

So I tried wpr.exe with the provided MsQuic.wprp file.

profile setting seems working, (wpr.exe -start MsQuic.wprp!Stacks.Light -filemode)
But I run wpr.exe -stop msquic.etl , it prints below. 

        Cannot change thread mode after it is set.

        Profile Id: RunningProfile

        Error code: 0x80010106

**Reply by nibanks** [reply]
For the `netsh` version you might need to either use cmd.exe instead of PowerShell or have to add some quotes (around "{...}").

As for the `wpr` issue, I've never seen that error before. I will have to ask some folks internally what that means.

**Reply by MinwooWebeng** [reply]
Thank you! I was able to take log with netsh in cmd. 

I exported it in a text format.
However, I'm having a hard time interpreting it.. 
[quic.txt](https://github.com/microsoft/msquic/files/13643395/quic.txt)

If you have some free time, may I ask you to take a look at this log?

+ I think this is the part where it failed.
[12]047C.5BDC::2023-12-12 11:07:25.190762000 [Microsoft-Quic][ lib] ERROR, 2147942402, RegOpenKeyExA failed. 
[12]047C.5BDC::2023-12-12 11:07:25.190769100 [Microsoft-Quic][cnfg][00000249AAE5BDD0] Failed to open app specific settings, 0x80070002 
[12]047C.5BDC::2023-12-12 11:07:25.190774800 [Microsoft-Quic][cnfg][0x249AAE5BDD0] Created, Registration=0x249AAE557A0

**Comment by MinwooWebeng** [comment, marked as answer]
Hey I just noticed that I also need .dll for the application to run! I didn't even think about it because it compiled with .lib, and didn't told me anything about dll while running.

Anyway, thanks for taking time to help me.
